protocols/autonat/src/behaviour.rs

```diff
@@ -478,35 +475,65 @@ impl NetworkBehaviour for Behaviour {
         self.inner.addresses_of_peer(peer)
     }
 
-    fn inject_event(
-        &mut self,
-        peer_id: PeerId,
-        conn: ConnectionId,
-        event: RequestResponseHandlerEvent<AutoNatCodec>,
-    ) {
-        self.inner.inject_event(peer_id, conn, event)
+    fn on_swarm_event(&mut self, event: FromSwarm<Self::ConnectionHandler>) {
+        match event {
+            FromSwarm::ConnectionEstablished(connection_established) => {
+                self.inner
+                    .on_swarm_event(FromSwarm::ConnectionEstablished(connection_established));
```

Comment:
Why only forward some of the `SwarmEvent`s to the underlying `libp2p-request-response` `NetworkBehaviour`? Why not forward all of them before the `match event {` above?